packages/client/components/Action/Action.tsx

```diff
@@ -33,53 +35,75 @@ const Action = memo(() => {
   useTrebuchetEvents()
   useServiceWorkerUpdater()
   const isInternalAuthEnabled = window.__ACTION__.AUTH_INTERNAL_ENABLED
+  // Global Banner
+  const isGlobalBannerEnabled = window.__ACTION__.GLOBAL_BANNER_ENABLED
+  const bannerText = window.__ACTION__.GLOBAL_BANNER_TEXT
+  const bannerBgColor = window.__ACTION__.GLOBAL_BANNER_BG_COLOR
+  const bannerColor = window.__ACTION__.GLOBAL_BANNER_COLOR
   return (
     <>
       <ErrorBoundary>
         <Snackbar />
         <Suspense fallback={<LoadingComponent spinnerSize={LoaderSize.WHOLE_PAGE} />}>
           <AnalyticsPage />
-          <Switch>
-            <Route exact path='/' render={(p) => <AuthenticationPage {...p} page={'signin'} />} />
-            <Route
-              exact
-              path={`/${SIGNIN_SLUG}`}
-              render={(p) => <AuthenticationPage {...p} page={'signin'} />}
-            />
-            <Route
-              exact
-              path={`/${CREATE_ACCOUNT_SLUG}`}
-              render={(p) => <AuthenticationPage {...p} page={'create-account'} />}
-            />
-            <Route exact path={`/auth/:provider`} component={AuthProvider} />
-            <Route path={`/saml-redirect`} component={SAMLRedirect} />
-            <Route
-              path='/retrospective-demo/:localPhaseSlug?/:stageIdxSlug?'
-              component={DemoMeeting}
-            />
-            <Route path='/retrospective-demo-summary/:urlAction?' component={DemoSummary} />
-            {isInternalAuthEnabled && (
-              <Route
-                exact
-                path={`/forgot-password`}
-                render={(p) => <AuthenticationPage {...p} page={'forgot-password'} />}
-              />
+          <div className='flex h-full max-h-screen flex-col'>
+            {isGlobalBannerEnabled && (
+              <GlobalBanner bgColor={bannerBgColor} color={bannerColor} text={bannerText} />
             )}
-            {isInternalAuthEnabled && (
-              <Route
-                path={`/forgot-password/submitted`}
-                render={(p) => <AuthenticationPage {...p} page={`forgot-password/submitted`} />}
-              />
-            )}
-            <Route
-              path='/verify-email/:verificationToken/:invitationToken?'
-              component={VerifyEmail}
-            />
-            <Route path='/reset-password/:token' component={SetNewPassword} />
-            <Route path='/team-invitation/:token' component={TeamInvitation} />
-            <Route path='/invitation-link/:token' component={InvitationLink} />
-            <Route component={PrivateRoutes} />
-          </Switch>
+            {/* Set up a container for the global banner and the main app area */}
+            <div
+              className={clsx(
+                'h-100 flex flex-1 flex-col overflow-hidden',
```

Comment:
can we get rid of everything here except the `pt-6`? having overflow-hidden as a parent could make things difficult because that means no component will be able to use the viewport scroll

Reply:
I think it was hard to control height of inner components on all the main views. In other words, I can do it on this one view and then tweak a few things here in there, or I have to go to most of the main views and adjust heights and what not. I’m not how much that would have to be conditional based on the `GLOBAL_BANNER_ENABLED` which I was trying to use less. So much of the app has to control inner scrolling. Maybe only mobile is relying on true viewport scrolling?

Reply:
I’ve got something I want to try but I think we should consider moving forward and making an additional set of improvements in a later PR.